src/components/popup/home/Balance.tsx

```diff
@@ -95,37 +125,43 @@ export default function Balance() {
 
   return (
     <BalanceHead>
-      {isLoading && <Loading style={{ width: "20px", height: "20px" }} />}
-      {!isLoading && (
-        <div style={{ display: "flex", flexDirection: "column", gap: "4px" }}>
-          <BalanceText onClick={() => setHideBalance((val) => !val)} noMargin>
-            {hideBalance ? (
-              <>{"*".repeat(totalFiatBalance.toFixed(2).length)}</>
-            ) : (
-              <NumberFlow
-                value={fiatBalance}
-                format={{
-                  style: "currency",
-                  currency: currency
-                }}
-              />
-            )}
-          </BalanceText>
-          <PriceChangeIndicator
-            percentageChange={percentage}
-            fiatChange={fiat.multipliedBy(percentage.dividedBy(100))}
-            hideBalance={hideBalance}
+      {isLoading ? (
+        <Loading
+          style={{
+            position: "absolute",
+            top: "50%",
+            left: "50%",
+            transform: "translate(-50%, -50%)",
+            width: "20px",
+            height: "20px"
+          }}
+        />
+      ) : null}
+
+      <BalanceWrapper $hideBalance={hideBalance || isLoading}>
+        <BalanceText onClick={() => setHideBalance((val) => !val)} noMargin>
+          <NumberFlow
```

Comment:
By keeping the `NumberFlow` element always rendered, we get nicer transitions when the price is loaded, hidden or shown. While loading or hidden, it now renders like this, and selection is not possible:

![image](https://github.com/user-attachments/assets/ff984678-f687-4f17-957a-55b8bf930ebc)